scripts/update_api.py

```diff
@@ -741,55 +741,97 @@ def mk_java(java_dir, package_name):
+        return 'external'
 
-Type2NapiBuilder = { VOID : '', VOID_PTR : '', INT : 'int32', UINT : 'uint32', INT64 : 'int64', UINT64 : 'uint64', DOUBLE : 'double',
+# Maps Z3 interface parameters to N-API value adapters
+Type2NapiBuilder = { VOID : '', VOID_PTR : '', INT : 'int32', UINT : 'uint32', INT64 : 'int64', UINT64 : 'int64', DOUBLE : 'double',
             FLOAT : 'float', STRING : 'string', STRING_PTR : 'array',
             BOOL : 'bool', SYMBOL : 'external', PRINT_MODE : 'int32', ERROR_CODE : 'int32' }
 
+def type2napi(t):
+    try:
+        return Type2Napi[t]
+    except KeyError:
+        return 'external'
+
 def type2napibuilder(t):
     try:
-       return Type2NapiBuilder[t]
-    except:
-       return "external"
+        return Type2NapiBuilder[t]
+    except KeyError:
+        return 'external'
 
+def param2napibuilder(p):
+    # TODO: Support in and out arrays and buffers
+    try:
+        k = param_kind(p)
+        if param_kind(p) == OUT:
+            raise Exception('Not Supported: See #')
+        elif k in [IN_ARRAY, INOUT_ARRAY, OUT_ARRAY]:
+            return 'array'
+        else:
+            return Type2NAPIBuilder[param_type(p)]
+    except:
+        # Effectively means this value is tracked by reference as a foreign object
+        return 'external'
 
 def mk_js(js_output_dir):
     with open(os.path.join(js_output_dir, "z3.json"), 'w') as ous:
        ous.write("{\n")
        ous.write("  \"api\": [\n")
+       first_outer = True
        for name, result, params in _dotnet_decls:
+           if first_outer:
+               first_outer = False
+           else:
+               ous.write("    },\n")
            ous.write("    {\n")
            ous.write("       \"name\": \"%s\",\n" % name)
-           ous.write("       \"c_type\": \"%s\",\n" % Type2Str[result])
+           ous.write("       \"c_type\": \"%s\",\n" % type2str(result))
            ous.write("       \"napi_type\": \"%s\",\n" % type2napi(result))                       
            ous.write("       \"arg_list\": [")
-           first = True
+           first_inner = True
            for p in params:
-               if first:
-                  first = False
+               if first_inner:
+                  first_inner = False
                   ous.write("\n         {\n")
                else:
                   ous.write(",\n         {\n")
-               t = param_type(p)
-               k = t
-               ous.write("            \"name\": \"%s\",\n" % "")                        # TBD
-               ous.write("            \"c_type\": \"%s\",\n" % type2str(t))
-               ous.write("            \"napi_type\": \"%s\",\n" % type2napi(t))        
-               ous.write("            \"napi_builder\": \"%s\"\n" % type2napibuilder(t))
+
+               """ The name of the param is only useful when debugging the bindings. Otherwise,
+               it is generated on the fly. """
+               ous.write("            \"name\": \"%s\",\n" % "")
```

Comment:
the names are absent in the meta-data. If the backend needs names, this code could cook up x1, x2, x3, etc.

Reply:
The generator already does that. What I meant to say in the comment was that it is useful to link those generated names with the names of the original parameters when debugging the API. But not strictly necessary. I'll clarify.